app/models/course.rb

```diff
@@ -164,6 +165,61 @@ def self.entry_requirement_options_without_nil_choice
 
   after_validation :remove_unnecessary_enrichments_validation_message
 
+  aasm column: :state do
+    state :new, initial: true
+    # validate has no enrichments
+    # validate has only "new" sites
+    state :draft
+    # validate has a draft enrichment
+    # validate has "new" or "running" sites
+    state :published
+    # validate has published enrichments and no draft enrichment
+    # validate has "running" sites
+    state :published_with_draft
+    # validate has published enrichments and a draft enrichment
+    # validate has "running" sites
+    state :withdrawn
+    # validate has "suspended" sites
+    state :rolled_over
+    # validate has no published enrichments
```

Comment:
so rollover....
has no published, so its a draft or a rollover single enrichment?
will it have no enrichment?
what of the sites set as?